src/client/views/cube-view/downloadable-dataset-context.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2021 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dataset, TabulatorOptions } from "plywood";
+import * as React from "react";
+import { ReactNode } from "react";
+import { Nullary, Unary } from "../../../common/utils/functional/functional";
+
+export interface DataSetWithTabOptions {
```

Comment:
I don't like naming here.  What do you think about:

DownloadableDataset {
  Dataset
  DatasetOptions or DownloadableDatasetOptions
}